### ISSUE TITLE
showing wrong data

### ISSUE THREAD

**Issue by kanthusb** [marketcalls/openalgo#678]
<img width="983" height="211" alt="Image" src="https://github.com/user-attachments/assets/157bdc45-f111-4dfa-a77e-80d06c75fd8a" />

im placed call options order using play ground its filled 126.65 pnl showing wrong 

<img width="998" height="116" alt="Image" src="https://github.com/user-attachments/assets/18429ae6-e586-4711-bd7a-dc110bdadd97" />

and in orderbook price showing 0  and can u please add live nifty and bank nifty in dashboard like broker     

and can u fix dashboard values refresh now we have to do manual refresh to see pnl and ltp

**Comment by marketcalls** [comment]
If you are sending a market order then price and trigger price are likely to be zero. If you are sending the limit order or Stop limit order then price and trigger price will show the pending level.

However if you want to see the executed price you need to see the tradebook. Not the orderbook which is meant to show only pending orders and market order with order status.

In our next phase we will be completely making the app with streaming data which is incoming shortly  with streaming PNL and realtime updates. Right now it is manual refresh.

**Comment by marketcalls** [comment]
Let me check the PNL calculation part

**Comment by kanthusb** [comment]
Thank you for replying and can you make a video on how to add strategy and configuration to in python and configure in strategy to configure multiple symbols right now I'm giving all in strategy one symbol

**Comment by kanthusb** [comment]
# -*- coding: utf-8 -*-

import time
import pandas as pd
from datetime import datetime, timedelta, time as dtime
from openalgo import api
import math

# ================= CONFIG =================
API_KEY = "PASTE_YOUR_OPENALGO_API_KEY_HERE"
HOST = "http://127.0.0.1:5000"

STRATEGY = "EMA_RSI_OPTIONS"
SPOT_EXCHANGE = "NSE"
OPTION_EXCHANGE = "NFO"
PRODUCT = "MIS"

EMA_LEN = 9
RSI_LEN = 14
ADX_LEN = 14
ADX_MIN = 20

LOT_SIZE = 75
STRIKE_STEP = 50  # NIFTY

# ---- TIME FILTER ----
ENTRY_START = dtime(9, 35)
ENTRY_END   = dtime(15, 15)

# ---- TARGET + SL + TSL ----
TARGET_POINTS = 20
SL_POINTS     = 20
TSL_STEP      = 10

client = api(api_key=API_KEY, host=HOST)

# ================= HELPERS =================
def atm_ce(price):
    return int(math.floor(price / STRIKE_STEP) * STRIKE_STEP)

def atm_pe(price):
    return int(math.ceil(price / STRIKE_STEP) * STRIKE_STEP)

def get_weekly_expiry():
    today = datetime.now()
    expiry_weekday = 1  # Tuesday
    days_ahead = expiry_weekday - today.weekday()
    if days_ahead < 0:
        days_ahead += 7
    expiry = today + timedelta(days=days_ahead)
    return expiry.strftime("%d%b%y").upper()

def build_option(symbol, strike, opt_type, expiry):
    return f"{symbol}{expiry}{strike}{opt_type}"

def calc_vwap(df):
    if "volume" not in df.columns:
        return df["close"]
    pv = (df["close"] * df["volume"]).cumsum()
    vol = df["volume"].cumsum().replace(0, 1)
    return pv / vol

def dynamic_itm_offset(adx):
    """
    Dynamic ITM logic:
    ADX >= 30  → ITM-3 (150 pts)
    ADX 20-29  → ITM-4 (200 pts)
    ADX < 20   → NO TRADE
    """
    if adx >= 30:
        return 3 * STRIKE_STEP
    elif adx >= 20:
        return 4 * STRIKE_STEP
    else:
        return None

# ================= STATE =================
position = None
entry_price = 0.0
target_price = 0.0
tsl_price = 0.0

print("EMA + RSI + VWAP OPTIONS STRATEGY STARTED (DYNAMIC ITM)")

# ================= MAIN LOOP =================
while True:
    try:
        # -------- FETCH SPOT DATA --------
        end_date = datetime.now().strftime("%Y-%m-%d")
        start_date = (datetime.now() - timedelta(days=5)).strftime("%Y-%m-%d")

        hist = client.history(
            symbol="NIFTY",
            exchange=SPOT_EXCHANGE,
            interval="15m",
            start_date=start_date,
            end_date=end_date
        )

        if not hist or "data" not in hist or not hist["data"]:
            time.sleep(15)
            continue

        df = pd.DataFrame(hist["data"])
        if len(df) < 30:
            time.sleep(15)
            continue

        # -------- INDICATORS --------
        df["ema"] = df["close"].ewm(span=EMA_LEN, adjust=False).mean()

        delta = df["close"].diff()
        gain = delta.clip(lower=0)
        loss = -delta.clip(upper=0)
        rs = gain.rolling(RSI_LEN).mean() / loss.rolling(RSI_LEN).mean().replace(0, 0.0001)
        df["rsi"] = 100 - (100 / (1 + rs))

        df["vwap"] = calc_vwap(df)

        # -------- ADX --------
        high, low, close = df["high"], df["low"], df["close"]

        plus_dm = high.diff()
        minus_dm = low.diff().abs()
        plus_dm = plus_dm.where((plus_dm > minus_dm) & (plus_dm > 0), 0.0)
        minus_dm = minus_dm.where((minus_dm > plus_dm) & (minus_dm > 0), 0.0)

        tr = pd.concat([
            high - low,
            (high - close.shift()).abs(),
            (low - close.shift()).abs()
        ], axis=1).max(axis=1)

        atr = tr.rolling(ADX_LEN).mean()
        plus_di = 100 * (plus_dm.rolling(ADX_LEN).mean() / atr)
        minus_di = 100 * (minus_dm.rolling(ADX_LEN).mean() / atr)
        dx = (abs(plus_di - minus_di) / (plus_di + minus_di).replace(0, 0.0001)) * 100
        df["adx"] = dx.rolling(ADX_LEN).mean()

        candle = df.iloc[-2]

        # -------- SIGNALS --------
        buy_signal = (
            candle.close > candle.ema and
            candle.rsi > 60 and
            candle.close > candle.vwap and
            candle.adx >= ADX_MIN
        )

        sell_signal = (
            candle.close < candle.ema and
            candle.rsi < 40 and
            candle.close < candle.vwap and
            candle.adx >= ADX_MIN
        )

        now = datetime.now().time()
        can_enter = ENTRY_START <= now <= ENTRY_END

        spot = candle.close
        expiry = get_weekly_expiry()

        # -------- ENTRY --------
        if position is None and can_enter:

            itmo = dynamic_itm_offset(candle.adx)
            if itmo is None:
                continue  # no trade in sideways market

            if buy_signal:
                atm = atm_ce(spot)
                strike = atm - itmo
                opt_symbol = build_option("NIFTY", strike, "CE", expiry)

            elif sell_signal:
                atm = atm_pe(spot)
                strike = atm + itmo
                opt_symbol = build_option("NIFTY", strike, "PE", expiry)
            else:
                continue

            entry_price = client.ltp(opt_symbol, OPTION_EXCHANGE)["ltp"]
            target_price = entry_price + TARGET_POINTS
            tsl_price = entry_price - SL_POINTS

            client.placesmartorder(
                strategy=STRATEGY,
                symbol=opt_symbol,
                action="BUY",
                exchange=OPTION_EXCHANGE,
                price_type="MARKET",
                product=PRODUCT,
                quantity=LOT_SIZE,
                position_size=LOT_SIZE
            )

            position = {"symbol": opt_symbol, "type": "CE" if buy_signal else "PE"}
            print(f"BUY {position['type']} {opt_symbol} @ {entry_price}")

        # -------- EXIT --------
        elif position is not None:
            ltp = client.ltp(position["symbol"], OPTION_EXCHANGE)["ltp"]

            if ltp - entry_price >= TSL_STEP:
                tsl_price = max(tsl_price, ltp - SL_POINTS)

            exit_cond = (
                ltp <= tsl_price or
                ltp >= target_price or
                (position["type"] == "CE" and sell_signal) or
                (position["type"] == "PE" and buy_signal)
            )

            if exit_cond:
                client.placesmartorder(
                    strategy=STRATEGY,
                    symbol=position["symbol"],
                    action="SELL",
                    exchange=OPTION_EXCHANGE,
                    price_type="MARKET",
                    product=PRODUCT,
                    quantity=LOT_SIZE,
                    position_size=0
                )

                print(f"EXIT {position['symbol']} @ {ltp}")
                position = None

    except Exception as e:
        print("ERROR:", e)

    time.sleep(15)
 now im using like this same analyze me testing strategy  is it possible to select symbols in startegy tab using 1 strategy multiple symboles

**Comment by marketcalls** [comment]
In the latest version the PNL Calculation is fixed. Consider upgrade to the latest openalgo version if not.

Consider reopen the ticket is you think the issue still persist
